src/hangar/dataset.py

```diff
@@ -257,70 +267,116 @@ def __delitem__(self, key):
+            name to assign to the same (assuming the dataset accepts named samples), by default None
 
         Returns
         -------
-        bool
-            if the operation was successful.
+        str
+            sample name of the stored data (assuming the operation was successful)
+
+        Raises
+        ------
+        ValueError
+            If no `name` arg was provided for dataset requiring named samples.
+        ValueError
+            If input data tensor rank exceeds specified rank of dataset samples.
+        ValueError
+            For variable shape datasets, if a dimension size of the input data
+            tensor exceeds specified max dimension size of the dataset samples.
+        ValueError
+            For fixed shape datasets, if input data dimensions do not exactally match
+            specified dataset dimensions.
+        TypeError
+            If type of `data` argument is not an instance of np.ndarray.
+        TypeError
+            If the datatype of the input data does not match the specifed data type of
+            the dataset
+        LookupError
+            If a data sample with the same name and hash value already exists in the
+            dataset.
         '''
-        if self._samples_are_named:
-            if not (isinstance(name, str)):
-                print('Error: dataset requires named samples')
-                return False
-        else:
-            try:
-                name = kwargs['bulk_add_name']
-            except KeyError:
-                name = parsing.generate_sample_name()
-
-        if self._schema_variable is True:
-            if data.ndim != len(self._schema_max_shape):
-                print(f'Error: rank of input tensor exceed rank of initialized tensor')
-                return False
-            for dataDimSize, schemaDimSize in zip(data.shape, self._schema_max_shape):
-                if dataDimSize > schemaDimSize:
-                    print('Error: dimensions of input data exceed max dimensions of dataset')
-                    return False
-        else:
-            if data.shape != self._schema_max_shape:
-                print('Error: shape of input tensor does not match shape of dataset schema.')
-                return False
 
-        if data.dtype.num != self._schema_dtype_num:
-            print('Error: data type of dataset and input data do not match')
-            return False
+        # ------------------------ argument type checking ---------------------
 
-        tmpconman = False if self._is_conman else True
-        if tmpconman:
-            self.__enter__()
+        try:
+            if not isinstance(data, np.ndarray):
+                msg = f'HANGAR TYPE ERROR:: `data` argument type: {type(data)} != `np.ndarray`'
+                raise TypeError(msg)
+
+            if self._samples_are_named:
+                if not (isinstance(name, str)):
+                    msg = f'HANGAR VALUE ERROR:: no `name` argument provided while '\
```

Comment:
IMO we should indicate the key should be string? Something like `HANGAR VALUE ERROR:: no name argument provided ... or provided value is not string` perhaps? Because this error is super confusing for somebody who makes an operation like 
```python
dset[1] = array
```

Reply:
Fixed!